### PR TITLE
Improve the java executable location logic, #241

### PR DESCRIPTION
**Pull request by starksm64**
Here is an initial update the the DevMojo logic for locating the java binary to address #241 . It allows for use of the maven toolchains as well as fallback to java.home and then the path. I have tested it on osx and windows.

### CODE REVIEW

**Review by cescoffier — APPROVED** [review]

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java [hunk 1]
```diff
@@ -114,6 +117,33 @@
...
+    @SuppressWarnings("UnusedDeclaration")
+    public void setToolchainManager(ToolchainManager toolchainManager) {
+        this.toolchainManager = toolchainManager;
+    }
+
+    public MavenSession getSession() {
+        return session;
+    }
+
+    @SuppressWarnings("UnusedDeclaration")
```
Comment (cescoffier) [inline devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java]: Not required.

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java [hunk 2]
```diff
@@ -114,6 +117,33 @@
...
+
+    @Component
+    private ToolchainManager toolchainManager;
+
+    public ToolchainManager getToolchainManager() {
+        return toolchainManager;
+    }
+
+    @SuppressWarnings("UnusedDeclaration")
+    public void setToolchainManager(ToolchainManager toolchainManager) {
```
Comment (cescoffier) [inline devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java]: Not required.

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java [hunk 3]
```diff
@@ -114,6 +117,33 @@
     @Parameter(defaultValue = "${preventnoverify}")
     private boolean preventnoverify = false;
 
+    /**
+     * The current build session instance.
+     */
+    @Parameter(defaultValue = "${session}", required = true, readonly = true)
+    private MavenSession session;
```
Comment (cescoffier) [inline devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java]: I believe we will have a conflict with another PR retrieving the session also (for a different reason).

**Review by cescoffier — APPROVED** [review]

**Comment by gsmet** [comment]
@starksm64 could you rebase? We merged a previous PR and they are conflicting on the `MavenSession` declaration. Thanks.

**Comment by stuartwdouglas** [comment]
This needs a rebase as the other PR got merged

**Comment by starksm64** [comment]
Working on rebase.

**Comment by starksm64** [comment]
So this is failing due to a formatting validation:

```
2019-03-13T14:06:23.3106300Z [ERROR] Failed to execute goal net.revelc.code.formatter:formatter-maven-plugin:2.8.1:validate (default) on project quarkus-maven-plugin: File '/home/vsts/work/1/s/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java' has not been previously formatted.  Please format file and commit before running validation! -> [Help 1]
```

I have the eclipse formatter plugin installed, not sure why the code was not formatted correctly. I just pushed a reformat update.

**Comment by starksm64** [comment]
Same problem, so I checked my formatting settings and found that it was not on this project. Trying again.

**Pull request merged**